### PR TITLE
Building Zip needed for Job submission from JobRequest. 

### PR DESCRIPTION
**Pull request by homezcx**
This PR is going to assemble all the files provided by JobRequest into a zip file and returns the path.

An example of the files assembled is assigned at HelloReef.cs line 69 around.

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs [hunk 1]
```diff
@@ -75,7 +81,9 @@ public void Submit(JobRequest jobRequest)
             var configModule = AzureBatchClientConfiguration.ConfigurationModule;
             string jobId = jobRequest.JobIdentifier;
             string commandLine = GetCommand(jobRequest.JobParameters);
-            _batchService.CreateJob(jobId, null, commandLine);
+            string jarPath = _jobJarMaker.CreateJobSubmissionJAR(jobRequest);
+            Uri blobUri = _storageUploader.UploadFile(new Uri(_azbatchFileNames.GetStorageJobFolder(jobId)), null).Result;
```
Comment (homezcx) [inline lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs:85]: FYI: IFile interface used in IStorageUploader might need refactored to support open file by path. Leave this to IStorageUploader PR.

Comment (ryumbra) [reply lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs:85]: yes, I took care of it.

File: lang/cs/Org.Apache.REEF.Client/AzureBatch/Util/JobJarMaker.cs [hunk 2]
```diff
@@ -43,7 +48,14 @@ internal sealed class JobJarMaker
...
         {
-            throw new NotImplementedException();
+            string localDriverFolderPath = CreateDriverFolder(jobRequest.JobIdentifier);
+            _driverFolderPreparationHelper.PrepareDriverFolder(jobRequest.AppParameters, localDriverFolderPath);
+            return _resourceArchiveFileGenerator.CreateArchiveToUpload(localDriverFolderPath);
+        }
+
+        private string CreateDriverFolder(string jobId)
+        {
+            return Path.GetFullPath(Path.Combine(Path.GetTempPath(), string.Join("-", "reef", jobId)));
```
Comment (ryumbra) [inline lang/cs/Org.Apache.REEF.Client/AzureBatch/Util/JobJarMaker.cs:58]: Is job id unique with every invocation? If not, we need to make sure that the directories are unique between each invocation.

**Review by ryumbra — APPROVED** [review]

File: lang/cs/Org.Apache.REEF.Client/AzureBatch/Util/JobJarMaker.cs (hunk 2 above)
Comment (homezcx) [reply lang/cs/Org.Apache.REEF.Client/AzureBatch/Util/JobJarMaker.cs:58]: yes. it is unique. Defined by "HelloworldJob-{datetime}"

**Pull request merged**